integration/tests/epochs/epoch_test.go

```diff
@@ -0,0 +1,79 @@
+package epochs
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+func TestEpochs(t *testing.T) {
+	suite.Run(t, new(Suite))
+}
+
+// TestViewsProgress asserts epoch state transitions over two full epochs
+// without any nodes joining or leaving.
+func (s *Suite) TestViewsProgress() {
```

Comment:
Not sure if you've had better luck, but I haven't been able to get this to pass locally so far.

Reply:
For me the test passes consistently in ~3min.
Looking at the logs you posted above it looks like there are some networking issues in your docker network:
```
{"level":"warn","node_role":"consensus","node_id":"ed17798ac60703e0edfa6ef8963a8b8c5038b0dc9d6f878bb9aa4fc2c0905b4e","component":"broker","dkg_instance_id":"dkg-flow-testnet-1","error":"could not get account details: could not get account: client: rpc error: cod
e = Unavailable desc = connection error: desc = \"transport: Error while dialing dial tcp 192.168.160.9:9000: connect: connection refused\"","time":"2021-06-25T00:03:24Z","message":"attempt 2/5 failed"}
```

Reply:
I ran the test 4 times on my machine. It passed once and showed symptoms of a race the other times (mismatch in [epoch counter](https://github.com/onflow/flow-go/blob/ed8ae9e8e06986367de8cd2be14870796a86d2e9/integration/tests/epochs/epoch_test.go#L80) or mismatch in [expected phase](https://github.com/onflow/flow-go/blob/ed8ae9e8e06986367de8cd2be14870796a86d2e9/integration/tests/epochs/epoch_test.go#L84)